qjackctl 0.4.1

david0 opened the pull request.

dunn commented on Library/Formula/qjackctl.rb:
```diff
@@ -0,0 +1,62 @@
+  depends_on "jack"
+  depends_on :x11
+
+  patch :DATA
```
Has this patch been submitted upstream?

dunn commented on Library/Formula/qjackctl.rb:
```diff
@@ -0,0 +1,62 @@
+
+  depends_on "qt"
+  depends_on "jack"
+  depends_on :x11
```
If this is a mandatory dependency, I'm afraid it'll have to go in our homebrew/x11 tap.

dunn commented:
We'll need a test also, thanks!

david0 commented:
I added another patch to remove the X11 dependency and added a test case.
All patches are already contained in upstream.

Im not sure whats going wrong at travis, but it looks like a problem with the travis setup in general.

mikemcquaid commented on Library/Formula/qjackctl.rb:
```diff
@@ -0,0 +1,118 @@
+  depends_on "qt"
+  depends_on "jack"
+
+  patch :DATA
```
Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

david0 replied:
Well this patch is a compilation of different commits that are already contained in upstream. The respective commit hashes are already contained in the patch. I will add additional comments why a backport is necessary from my point of view.

mikemcquaid commented on Library/Formula/qjackctl.rb:
```diff
@@ -0,0 +1,118 @@
+
+    system "make"
+    prefix.install "src/qjackctl.app"
+    man1.install "qjackctl.1"
```
Is there a `make install` you can use?

david0 replied:
There is a make install, but it is not of much help: it installs the app to ``/usr/local/Cellar/qjackctl/0.4.1/bin/qjackctl.app/`` which is not what we want I guess.
Also when using make install, I don't know how to make homebrew show up the ``Run `brew linkapps qjackctl` to symlink these to /Applications.``-message

dunn replied:
`linkapps` only works when the app bundle is installed in `libexec` or `prefix`.

david0 commented:
I updated the formula once again to use make install and added comments to justificate the paches.

mikemcquaid commented on Library/Formula/qjackctl.rb:
```diff
@@ -0,0 +1,135 @@
+  head "http://git.code.sf.net/p/qjackctl/code", :using=>:git
+
+  stable do
+    patch :DATA
```
Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

david0 replied:
No its not possible to submit this patch, I'm sorry. All changesets in this patch are contained in upstream, as i already said before. Their commit hashes are contained in the respective part and can be used to verify this change in upstream.  
And there are also comments explaining their necessity (in the patch data part)

> Am 24.11.2015 um 11:25 schrieb Mike McQuaid <notifications@github.com>:
> 
> In Library/Formula/qjackctl.rb:
> 
> > @@ -0,0 +1,135 @@
> > +class Qjackctl < Formula
> > +  desc "simple Qt application to control the JACK sound server daemon"
> > +  homepage "http://qjackctl.sourceforge.net"
> > +  url "https://downloads.sourceforge.net/qjackctl/qjackctl-0.4.1.tar.gz"
> > +  sha256 "98024e1ea9f55ac16c290feda051d6494b2261e83f6a918f0fa583b509e2bfd1"
> > +  head "http://git.code.sf.net/p/qjackctl/code", :using=>:git
> > +
> > +  stable do
> > +    patch :DATA
> Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!
> 
> —
> Reply to this email directly or view it on GitHub.
>

mikemcquaid replied:
In that case add a link to the upstream changesets and a Ruby comment above here explaining why the patch is needed, thanks.

mikemcquaid commented on Library/Formula/qjackctl.rb:
```diff
@@ -0,0 +1,135 @@
+  end
+
+  test do
+    Kernel.system "qjackctl", "--version"
```
Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

Also, instead of using `Kernel.system` use `shell_output` with `1` as the second parameter to verify the exit code and verify the command output with `assert_match`

david0 commented:
I updated the test case and added some comments based on the review.

mikemcquaid commented on Library/Formula/qjackctl.rb:
```diff
@@ -0,0 +1,142 @@
+  stable do
+    # fixes compile error with getDeviceUIDFromID and combo box device selection is general
+    # also fixes linking without X11
+    patch :DATA
```
Please add the links here too.

david0 replied:
Done

mikemcquaid commented:
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!
